packages/sanity/src/core/store/key-value/KeyValueStore.ts

```diff
@@ -1,18 +1,19 @@
+import {type SanityClient} from '@sanity/client'
 import {merge, type Observable, Subject} from 'rxjs'
 import {filter, map, switchMap} from 'rxjs/operators'
 
-import {resolveBackend} from './backends/resolve'
+import {serverBackend} from './backends/server'
 import {type KeyValueStore, type KeyValueStoreValue} from './types'
 
 /** @internal */
-export function createKeyValueStore(): KeyValueStore {
-  const storageBackend = resolveBackend()
```

Comment:
@jtpetty has recommended we fall back to localStorage if the API is not available here. I've not included it in this PR since I didn't find a trivial way to deal with the asynchronous state of pinging the backend from within the resourceCache context. Hopefully it's possible.